src/components/instantlaunches/index.js

```diff
@@ -107,6 +110,16 @@ const InstantLaunchButton = ({
     const [open, setOpen] = React.useState(false);
     const [ilUrl, setIlUrl] = React.useState();
     const theme = useTheme();
+    const [bootstrapInfo] = useBootstrapInfo();
+    const homePath = useHomePath();
+
+    const preferences = bootstrapInfo?.preferences;
+
+    const output_dir =
+        preferences?.default_output_folder?.path ||
+        preferences?.system_default_output_dir?.path ||
+        (homePath && `${homePath}/analyses`) ||
+        "";
```

Comment:
This looks like a duplicate of the logic already in the apps launch form:
https://github.com/cyverse-de/sonora/blob/f067afa56fc0f435124955734e96637ca516d2a0/src/components/apps/launch/index.js#L74-L80

I would refactor this logic into a new hook, maybe named something like `useDefaultOutputDir`in `src/components/data/utils.js`, similar to `useSelectorDefaultFolderPath`:
https://github.com/cyverse-de/sonora/blob/f067afa56fc0f435124955734e96637ca516d2a0/src/components/data/utils.js#L167-L176

Then both this file and `src/components/apps/launch/index.js` would no longer have to reference `useBootstrapInfo` or `useHomePath`, since they are only using them to get this default output folder path.

Reply:
Thanks for the heads up on this, I completely forgot that I had copied that code into here. That's what I get for letting a PR sit for a month...

Reply:
Okay, should be updated.